contracts/twap/CumulativeTwap.sol

```diff
@@ -125,42 +125,55 @@ contract CumulativeTwap is BlockContext {
         view
         returns (Observation memory beforeOrAt, Observation memory atOrAfter)
     {
-        uint8 index = currentObservationIndex;
-        uint8 beforeOrAtIndex;
-        uint8 atOrAfterIndex;
+        beforeOrAt = observations[currentObservationIndex];
 
-        // run at most 256 times
-        uint256 observationLen = observations.length;
-        uint256 i;
-        for (i = 0; i < observationLen; i++) {
-            if (observations[index].timestamp <= targetTimestamp) {
-                // if the next observation is empty, using the last one
-                // it implies the historical data is not enough
-                if (observations[index].timestamp == 0) {
-                    atOrAfterIndex = beforeOrAtIndex = index + 1;
-                    break;
-                }
-                beforeOrAtIndex = index;
-                atOrAfterIndex = beforeOrAtIndex + 1;
-                break;
-            }
-            index--;
+        // if the target is chronologically at or after the newest observation, we can early return
+        if (observations[currentObservationIndex].timestamp <= targetTimestamp) {
+            return (beforeOrAt, beforeOrAt);
```

Comment:
Not sure why this is different from Uni's implementation (even in https://github.com/perpetual-protocol/perp-oracle-contract/pull/63/files, it's still different: there's no if else conditions)